app/controllers/api/v1/statuses_controller.rb

```diff
@@ -114,4 +114,9 @@ def status_params
   def pagination_params(core_params)
     params.permit(:limit).merge(core_params)
   end
+
+  def authorize_if_hidden
+    status = Status.find(params[:id])
+    doorkeeper_authorize! :read if status.hidden?
```

Comment:
If I'm not mistaken, current behaviour is to pretend to 404 when trying to view a status you don't have a permission to view - because we don't want to reveal the *existence* of such a status. Now this breaks that behaviour - if you can't view a status without credentials, but with credentials it throws 404, you know that it exists.

Instead, if this method is to be public, it should always be public.